packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1681,20 +1775,39 @@ export function catalogServiceBuilder(
   };
 }
 
-const isUserAllowedToSeeDraft = (
+async function isUserAllowedToSeeDraft(
+  eservice: EService,
   authData: AuthData,
-  producerId: TenantId
-): boolean =>
-  hasPermission(
-    [userRoles.ADMIN_ROLE, userRoles.API_ROLE, userRoles.SUPPORT_ROLE],
-    authData
-  ) && authData.organizationId === producerId;
-
-const applyVisibilityToEService = (
+  readModelService: ReadModelService
+): Promise<boolean> {
+  if (
+    !hasPermission(
+      [userRoles.ADMIN_ROLE, userRoles.API_ROLE, userRoles.SUPPORT_ROLE],
+      authData
+    )
+  ) {
+    return false;
+  }
+
+  if (authData.organizationId === eservice.producerId) {
+    return true;
+  }
+
+  const activeDelegation = await readModelService.getLatestDelegation({
+    eserviceId: eservice.id,
+    delegateId: authData.organizationId,
+    states: [delegationState.active],
+  });
+
+  return Boolean(activeDelegation);
```

Comment:
I'm not getting this check. The method name is `isUserAllowedToSeeDraft` but here you are looking for the latest delegation with active state? So the logic is: if the `eservice` has been already delegated to you you can see the draft version?

Reply:
With the `delegateId: authData.organizationId` filter we are checking if the caller is the delegate, if any. 
So, the logic should be: if the you, the caller, are the delegate of the e-service, you can see the draft version